libyul/backends/evm/SSAControlFlowGraphBuilder.cpp

```diff
@@ -535,8 +540,22 @@ void SSAControlFlowGraphBuilder::operator()(Leave const& _leaveStatement)
 void SSAControlFlowGraphBuilder::operator()(Block const& _block)
 {
 	ScopedSaveAndRestore saveScope(m_scope, m_info.scopes.at(&_block).get());
-	for (auto const& statement: _block.statements)
-		std::visit(*this, statement);
+	if (m_currentBlock == m_graph.entry)
```

Comment:
There is no guarantee that all function definitions are in the outermost block.

Reply:
(Scoping rules make sure that the only *visible* function definitions are in the current block or in blocks further out, but any block can contain further function definitions)

Reply:
That also means that function definitions can contain nested function definitions in principle, which won't work as things are constructed right now, will it?

It may be better to move the ``buildFunctionGraph`` thing to the ``FunctionDefinition`` visit instead. That means carrying around a reference to the ``ControlFlow`` thing, but I don't see a problem with that.
Then you could also fetch ``canContinue`` out of the called functions subgraph on function calls.

Reply:
As in to be precise: whenever hitting a ``Block``, you first need to look for ``FunctionDefinition``s in it and build graphs for those, and only then visit the rest of the block - but that has to happen for *all* blocks, including blocks in other functions or arbitrarily nested blocks. We do run some optimizations pulling the nested functions out by default these days, but that can be overridden by custom settings, which would violate that assumption.